src/components/TagContainer.ts

```diff
@@ -0,0 +1,203 @@
+        return createElement(Tag, {
+            alertMessage: this.state.alertMessage,
+            tagStyle: this.props.tagStyle,
+            className: this.props.class,
+            createTag: this.createTag,
+            enableCreate: this.props.enableCreate,
+            enableSuggestions: this.props.enableSuggestions,
+            inputPlaceholder: this.props.inputPlaceholder,
+            lazyLoad: this.props.lazyLoad,
+            lazyLoadTags: this.lazyLoadTags,
+            readOnly: this.isReadOnly(),
+            showError: this.showErrorMessage,
+            style: TagContainer.parseStyle(this.props.style),
+            tagLimit: this.props.tagLimit,
+            tagLimitMessage: this.props.tagLimitMessage,
+            tags: this.state.tags,
+            tagValue: this.state.tag
+        });
+    }
+
+    componentWillReceiveProps(newProps: TagContainerProps) {
+        if (newProps.mxObject !== this.props.mxObject) {
+            if (!this.props.lazyLoad) { this.fetchCurrentTags(newProps.mxObject); }
+            this.resetSubscriptions(newProps.mxObject);
+        }
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+    }
+
+    private showErrorMessage(message: string) {
+        window.mx.ui.error(message);
+    }
+
+    public static parseStyle(style = ""): {[key: string]: string} {
+        try {
+            return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            // tslint:disable-next-line no-console
+            console.log("Failed to parse style", style, error);
+        }
+
+        return {};
+    }
+
+    private getValue(attribute: string, mxObject?: mendix.lib.MxObject): string {
+        return mxObject ? (mxObject.get(attribute) as string) : "";
+    }
+
+    private isReadOnly() {
+        const { tagAttribute, editable, mxObject, readOnly } = this.props;
+        if (editable === "default" && mxObject) {
+            return readOnly || mxObject.isReadonlyAttr(tagAttribute);
+        }
+
+        return true;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                attr: this.props.tagAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
```

Comment:
Missing the total flow for validation feedback. Please implement. like 
https://github.com/mendixlabs/boolean-slider/blob/master/src/components/SwitchContainer.ts#L128-L132
Show the error inline with a nice alert (no popup)

Reply:
validation feedback Added